pkg/controller/common/settings/replace_test.go

```diff
@@ -0,0 +1,71 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package settings
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+var testReplacement = Replacement{
+	Path: []string{
+		"node", "roles",
+	},
+	Expected:    nil,
+	Replacement: make([]string, 0),
+}
+
+func TestCanonicalConfig_RenderSeq(t *testing.T) {
+
+	tests := []struct {
+		name     string
+		input    map[string]interface{}
+		expected []byte
+	}{
+		{
+			name: "nil roles replacement",
+			input: map[string]interface{}{
+				"node": map[string]interface{}{
+					"roles": make([]string, 0),
```

Comment:
Should this be `roles: nil`? Or maybe add a separate test to validate if the "[] to nil" behavior is still there in ucfg?